swift-codec/src/main/java/com/facebook/swift/codec/metadata/ThriftUnionMetadataBuilder.java

```diff
@@ -166,6 +170,34 @@ protected boolean isValidateSetter(Method method)
         return method.getParameterTypes().length == 1;
     }
 
+    private void checkFieldRequiredness()
+    {
+        Set<Short> visitedInvalidFields = Sets.newHashSet();
+
+        for (FieldMetadata field : fields) {
+            if (field.getType() == THRIFT_UNION_ID) {
+                continue;
+            }
+
+            switch (field.getRequiredness()) {
+                case REQUIRED:
+                case OPTIONAL:
+                    if (!visitedInvalidFields.contains(field.getId())) {
+                        metadataErrors.addError(
+                                "Thrift union '%s' field '%s(%s)' should not be marked required or optional",
```

Comment:
If we fix that (so exported swift -> IDL won't break) then I'd actually like to change this. I almost changed it while working on this until I checked fbcode thrift compiler. I'll put a comment to this effect.